include/seqan3/alignment/pairwise/policy/scoring_scheme_policy.hpp

```diff
@@ -0,0 +1,58 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::scoring_scheme_policy.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/core/platform.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief The CRTP-policy that stores the scoring scheme used for this alignment algorithm.
+ * \ingroup alignment_policy
+ * \tparam alignment_algorithm_t The derived type (seqan3::detail::alignment_algorithm) to be augmented with this
+ *                               CRTP-policy.
+ * \tparam scoring_scheme_t The type of the scoring scheme.
+ *
+ * \remarks The template parameters of this CRTP-policy are selected in the
+ *          seqan3::detail::alignment_configurator::configure_scoring_scheme when selecting the alignment for the given
+ *          configuration.
+ */
+template <typename alignment_algorithm_t, typename scoring_scheme_t>
+class scoring_scheme_policy
```

Comment:
All policies are CRTP-base classes, which means there constructors are explicitly declared private and can only be instantiated through the derived class. Hence, they all get there constructors defined since an aggregate cannot have private constructors.